### PR TITLE
remove andThen overload, #25133

### PR DESCRIPTION
**Pull request by patriknw**
Refs #25133

### CODE REVIEW

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala [hunk 1]
```diff
@@ -140,30 +140,30 @@ object PersistentBehaviorSpec {
...
-            .andThen {
+            .andThen { _ ⇒
               loggingActor ! firstLogging
             }
 
         case LogThenStop ⇒
-          Effect.none
-            .andThen {
+          Effect.none[Event, State]
+            .andThen { _ ⇒
```
Comment (raboof) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala]: These 2 examples, with `Effect.none.andThen`, look quite strange to me.

Can't/shouldn't we just do `case LogThenStop => { loggingActor ! firstLogging; Effect.stop }`?

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala [hunk 2]
```diff
@@ -140,30 +140,30 @@ object PersistentBehaviorSpec {
         case IncrementTwiceAndThenLog ⇒
           Effect
             .persist(Incremented(1), Incremented(1))
-            .andThen {
+            .andThen { (_: State) ⇒
```
Comment (raboof) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala]: Needed here as well?

**Review by raboof — COMMENTED** [review]
:+1:, 1 semi-related comment about the `Effect.none.andThen` example

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala]: yes, compiler says so:
```
[error] /Users/patrik/dev/akka/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala:163:24: missing parameter type
[error]             .andThen { _ ⇒
[error]                        ^
```

If it wasn't for the two `.andThen` it wouldn't be needed here.

One would never write code like this, but we are exercising the edge cases.

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala]: sure, but the purpose of these tests are to exercise the edge cases of chaining the callbacks, I presume.

**Comment by patriknw** [comment]
rebased

**Comment by patriknw** [comment]
PLS BUILD
